retrieval/scrape_test.go

```diff
@@ -875,11 +876,11 @@ type errorAppender struct {
 func (app *errorAppender) Add(lset labels.Labels, t int64, v float64) (string, error) {
 	switch lset.Get(model.MetricNameLabel) {
 	case "out_of_order":
-		return "", storage.ErrOutOfOrderSample
+		return "", tsdb.ErrOutOfOrderSample
 	case "amend":
-		return "", storage.ErrDuplicateSampleForTimestamp
+		return "", tsdb.ErrAmendSample
 	case "out_of_bounds":
-		return "", storage.ErrOutOfBounds
+		return "", tsdb.ErrOutOfBounds
```

Comment:
Oh wow, sorry. I missed this. We do make the conversion at the shim layer already: https://github.com/prometheus/prometheus/blob/dev-2.0/storage/tsdb/tsdb.go#L136-L141